### PR TITLE
Add support for data_files #890

### PR DESCRIPTION
**Pull request by delphyne**
# Pull Request Check List

- [x] Added **tests** for changed code.
- [x] Updated **documentation** for changed code.

Adds support for [data_files](https://docs.python.org/2/distutils/setupscript.html#installing-additional-files) to poetry.

Addresses enhancement request #890 

*Note:* Black made a number of formatting changes in files that I touched, but in sections I wasn't editing.

### CODE REVIEW

**Comment by delphyne** [comment]
I'll fix the test failures first thing tonight.  The source of the problem is that maps retain insertion order as of python 3.6, but not in 2.7 & 3.5.

**Comment by delphyne** [comment]
@sdispater are there any additional tests or things you'd like addressed in this PR?

**Comment by mtkennerly** [comment]
@delphyne, for what it's worth, I submitted a similar PR a few months ago, #539. I haven't compared them in detail, but it may be worth incorporating some things from mine (e.g., mine includes changes to poetry-schema.json). It looks like mine has some conflicts now, so yours is probably a better base since it's closer to the current code.

**Comment by delphyne** [comment]
@mtkennerly that's unfortunate!  Also a little unfortunate that your PR has been hanging out for so long without any comment.  I tried to search for open issues and PRs for this functionality, but I didn't find your PR, so sorry for the duplicated effort.  

What is the purpose of the poetry-schema file? all the tests pass without the changes, and it's not mentioned in the contributing docs.

**Comment by thejohnfreeman** [comment]
@sdispater do you have help with this project? Seems like you might be overwhelmed with 76 outstanding pull requests and 422 open issues as of this comment. Do you need or want help? Who do you trust to perform code reviews?

File: docs/docs/pyproject.md [hunk 1]
```diff
@@ -133,6 +133,59 @@ packages = [
     Poetry is clever enough to detect Python subpackages.
 
     Thus, you only have to specify the directory where your root package resides.
+    
+## data_files
+
+A list of files to be installed using the [data_files](https://docs.python.org/2/distutils/setupscript.html#installing-additional-files)
```
Comment (brycedrennan) [inline docs/docs/pyproject.md]: Lets link to the [Python 3 version](https://docs.python.org/3/distutils/setupscript.html#installing-additional-files)?

File: poetry/masonry/builders/builder.py [hunk 2]
```diff
@@ -40,6 +40,7 @@ def __init__(self, poetry, env, io):
             self._path.as_posix(),
             packages=self._package.packages,
             includes=self._package.include,
+            data_files=self._package.data_files,
```
Comment (brycedrennan) [inline poetry/masonry/builders/builder.py]: How does this compare to 
https://github.com/sdispater/poetry/pull/539/files#diff-adffe92a131c3e1bf7e56fb29b1bc265R134

File: poetry/packages/project_package.py [hunk 3]
```diff
@@ -14,6 +14,7 @@ def __init__(self, name, version, pretty_version=None):
         self.packages = []
         self.include = []
         self.exclude = []
+        self.data_files = {}
```
Comment (brycedrennan) [inline poetry/packages/project_package.py]: why a `{}` vs `[]` as compared to https://github.com/sdispater/poetry/pull/539/files#diff-79da47ec0c1b34186a766622ab6e78ffR17

**Comment by abn** [comment]
This should now be covered by https://python-poetry.org/docs/pyproject/#include-and-exclude. If this is not the case, please feel free to rebase and open a new PR.

**Comment by fish-face** [comment]
@delphyne / @mtkennerly are you able to update/recreate this PR? it's still relevant, as per the issue discussion.

**Comment by fish-face** [comment]
Yes, that's very reasonable! Well consider this my endorsement anyway, for
whatever that's worth to anyone!

**Pull request closed without merging**